### PR TITLE
Update crds application connector

### PR DESCRIPTION
**Pull request by werdes72**
application connector crds

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/11170

### CODE REVIEW

**Comment by strekm** [comment]
@janmedrek can you also handle `components/compass-runtime-agent/internal/compassconnection/testdata/compass-connection.crd.yaml` in this PR?

File: installation/resources/crds/application-connector/applications.applicationconnector.crd.yaml [hunk 1]
```diff
@@ -1,108 +1,113 @@
...
+          type: object
           properties:
-            compassMetadata:
-              type: object
-              required:
-              - "authentication"
+            spec:
               properties:
-                authentication:
+                compassMetadata:
```
Comment (everesio) [inline installation/resources/crds/application-connector/applications.applicationconnector.crd.yaml:19]: The issue was suppose to change resource version not CRD API

Comment (everesio) [reply installation/resources/crds/application-connector/applications.applicationconnector.crd.yaml:19]: @werdes72 If CRD needs API changes this should be done in different PR. Additionally CRD update is incompatible which required new CRD version and application refactoring

Comment (everesio) [reply installation/resources/crds/application-connector/applications.applicationconnector.crd.yaml:19]: Sorry I was confused by the git diff, there are no API changes

Comment (werdes72) [reply installation/resources/crds/application-connector/applications.applicationconnector.crd.yaml:19]: Don't worry, me too, until I checked those changes locally.

**Review by akgalwas — APPROVED** [review]

**Review by janmedrek — DISMISSED** [review]

**Review by crabtree — APPROVED** [review]

**Review by akgalwas — DISMISSED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by janmedrek — APPROVED** [review]

**Comment by ksputo** [comment]
Please also update documentation and YAML snippets, for example:

https://github.com/kyma-project/kyma/blob/main/docs/application-connector/08-08-application-mapping.md
https://github.com/kyma-project/kyma/blob/main/components/application-broker/contrib/examples/selective-mapping.yaml

https://github.com/kyma-project/kyma/search?l=Markdown&q=ApplicationMapping

**Comment by everesio** [comment]
@ksputo The version of the CRD was not changed, but only the version of the container i.e. apiextention. This means that no further changes are required

**Review by ksputo — APPROVED** [review]
lgtm

**Review by strekm — APPROVED** [review]
/meow

**Review by crabtree — APPROVED** [review]

**Pull request merged**